### DISCUSSION TITLE
technology/docker/build-docker-from-scratch-based-on-deboostrap/

### DISCUSSION THREAD

**Discussion by giscus** [songtianlun/songtianlun.github.io#7] [Announcements]
# technology/docker/build-docker-from-scratch-based-on-deboostrap/

基于 debootstrap 从零生成最小 debian 并制作 docker 镜像.

https://www.frytea.com/technology/docker/build-docker-from-scratch-based-on-deboostrap/

**Comment by charliemini** [comment]
尝试用这个步骤制作uos基础镜像了，发现所有deb都没装上，发现是dpkg的问题，/var/lib/dpkg/status需要修改，手动安装deb。但是这个1010这个源构成的系统镜像还需要激活，其它源又报错没有release文件。最后这个镜像也不能作为申威上能在生产环境中使用的docker镜像，不知道你后续还有研究么？

**Reply by songtianlun** [reply]
后续没有研究了，但从最近对国产系统的体验来看，软件包的上下兼容性应该是村在很大问题的。就你的描述， UOS 的软件源大概是后续更新引入了一些问题导致目前不可用，这个方法跟软件源质量高度相关，如果软件源有问题就很难解决了。

这里建议尝试直接用官方提供的申威 iso 安装系统后自行裁剪制作成 docker 来用吧，或许比这个方法靠谱些。
